zerver/webhooks/facebook/view.py

```diff
@@ -0,0 +1,123 @@
+    return message
+
+def page_object(payload: Dict[Text, Any]) -> Text:
+    field = payload['entry']['changes']['field']
+
+    message = ''
+    if field == 'conversations':
+        message = message + 'Updated **conversations**'
+        message = message + '\n[Open conversations...](https://www.facebook.com/' \
+                            '{_page_id}/{_thread_id})'.format(
+                                _page_id=payload['entry']['changes']['value']['page_id'],
+                                _thread_id=payload['entry']['changes']['value']['thread_id']
+                            )
+    elif field == 'website':
+        message = message + 'Changed **website**'
+    return message
+
+def permissions_object(payload: Dict[Text, Any]) -> Text:
+    field = payload['entry']['changes']['field']
+    message = '**{_field} permission** changed'.format(_field=field)
+    if field == 'ads_management':
+        message = message + '\n* {_verb}'.format(
+            _verb=payload['entry']['changes']['value']['verb']
+        )
+        for id in payload['entry']['changes']['value']['target_ids']:
+            message = message + '\n  * {_id}'.format(_id=id)
+    elif field == 'manage_pages':
+        message = message + '\n* {_verb}'.format(
+            _verb=payload['entry']['changes']['value']['verb']
+        )
+        for id in payload['entry']['changes']['value']['target_ids']:
+            message = message + '\n  * {_id}'.format(_id=id)
+    return message
+
+def application_object(payload: Dict[Text, Any]) -> Text:
+    field = payload['entry']['changes']['field']
+    message = '**{_field}** received'.format(_field=field)
+    if field == 'plugin_comment':
+        message = message + '\n**{_msg_user}:**\n```quote\n{_message}\n```'.format(
+            _msg_user=payload['entry']['changes']['value']['from']['name'],
+            _message=payload['entry']['changes']['value']['message']
+        )
+    if field == 'plugin_comment_reply':
+        message = message + '\n**{_prt_msg_user}:** (Parent)\n' \
+                '```quote\n{_prt_message}\n```'.format(
+                    _prt_msg_user=payload['entry']['changes']['value']['parent']['from']['name'],
+                    _prt_message=payload['entry']['changes']['value']['parent']['message']
+                )
+        message = message + '\n**{_cld_msg_user}:**\n```quote\n' \
+                '```quote\n{_cld_message}\n```\n```'.format(
+                    _cld_msg_user=payload['entry']['changes']['value']['from']['name'],
+                    _cld_message=payload['entry']['changes']['value']['message']
+                )
+    return message
+
+@api_key_only_webhook_view("Facebook")
+@has_request_variables
+def api_facebook_webhook(request: HttpRequest, user_profile: UserProfile,
+                        stream: Text=REQ(default='Facebook'), token: Text=REQ(),
+                        payload: Dict[Text, Any]=REQ(argument_type='body')) -> HttpResponse:
+    tmp_body = str(payload)
```

Comment:
I don't think you need to convert this to a `str`.

Reply:
If I remove this converting, script make error as cannot edit Dict Object.

Is there any other way?